common/config/environment.go

```diff
@@ -136,3 +136,11 @@ func (e *CommonEnvironment) GetObjectWithDefault(config *sdkconfig.Config, param
 
 	return defaultValue
 }
+
+type Environment interface {
```

Comment:
I'm not sure this belongs here.
I think what you want is a `Environment` interface with the methods you're interested in, probably in `vm`.

Reply:
I put this interface here because of dependencies.

`Environment` is used by both `common/vm` and `common/os` and the first depends on the later. `Environment` cannot be in `common/vm`.  `Environment` also depends on `CommonEnvironment` and so cannot be in the root package `common`.
`Environment` can be in `config/config` or in a dedicated package.
If you prefer, I can rename `common/config/environment.go` to `common/config/commonEnvironment.go` and move the code of `Environment` inside `common/config/environment.go`.

Reply:
Given the links between `vm` and `os`, it may mean that it should be a single package.
Maybe another option is to have something like `common/vm/environment.go` and then `common/vm/builder/*` with existing code.

Reply:
As I have several local PRs that are based on this file hierarchy, I will do the change in a later PR. Tracked by `AETT-57`.